src/main/java/com/nike/cerberus/cache/DatabaseCache.java

```diff
@@ -0,0 +1,237 @@
+  private MetricsService metricsService;
+  private Integer repeatReadThreshold;
+
+  protected static final String GLOBAL_DATA_TTL_IN_SECONDS = "cms.mybatis.cache.global.dataTtlInSeconds";
+  protected static final String DATA_TTL_IN_SECONDS_OVERRIDE_PATH_TEMPLATE = "cms.mybatis.cache.%s.dataTtlInSeconds";
+  protected static final String GLOBAL_REPEAT_READ_COUNTER_EXPIRE_IN_SECONDS = "cms.mybatis.cache.global.repeatReadCounterResetInSeconds";
+  protected static final String REPEAT_READ_COUNTER_EXPIRE_IN_SECONDS_OVERRIDE_PATH_TEMPLATE = "cms.mybatis.cache.%s.repeatReadCounterResetInSeconds";
+  protected static final String GLOBAL_REPEAT_READ_THRESHOLD = "cms.mybatis.cache.global.repeatReadThreshold";
+  protected static final String REPEAT_READ_THRESHOLD_OVERRIDE_PATH_TEMPLATE = "cms.mybatis.cache.%s.repeatReadThreshold";
+  protected static final int DEFAULT_GLOBAL_DATA_TTL_IN_SECONDS = 10;
+  protected static final int DEFAULT_REPEAT_READ_COUNTER_EXPIRE_IN_SECONDS = 2;
+  protected static final int DEFAULT_REPEAT_READ_THRESHOLD = 2;
+
+  protected final String id;
+  protected com.github.benmanes.caffeine.cache.Cache<Object, Object> dataCache;
+  protected com.github.benmanes.caffeine.cache.Cache<Object, Counter> autoExpiringRepeatReadCounterMap;
+  protected Counter hitCounter;
+  protected Counter missCounter;
+
+  public DatabaseCache(String id) {
+    this.id = id;
+  }
+
+  /**
+   * This method gets called after this class is instantiated by MyBatis and all the properties have been set.
+   */
+  @Override
+  public void initialize() {
+    // Util we can get the MyBatis Guice Module updated, this is our best bet, for getting the Guice instances.
+    // https://groups.google.com/forum/#!msg/mybatis-user/Ekd1LTNVIDc/t2xGuvETBgAJ
+    Injector injector = StaticInjector.getInstance();
+
+    Config config = injector.getInstance(Config.class);
+    metricsService = injector.getInstance(MetricsService.class);
+
+    String mapperKey = StringUtils.uncapitalize(id.replaceFirst("com.nike.cerberus.mapper.", ""));
+    int expireTimeInSeconds = getExpireTimeInSeconds(config, mapperKey);
+    int counterExpireTimeInSeconds = getRepeatReadCounterExpireTimeInSeconds(config, mapperKey);
+    repeatReadThreshold = getRepeatReadThreshold(config, mapperKey);
+
+    log.info("Database cache created with mapperKey: {}, expireTimeInSeconds: {}, counterExpireTimeInSeconds: {}, repeatReadThreshold: {}",
+        mapperKey, expireTimeInSeconds, counterExpireTimeInSeconds, repeatReadThreshold);
+
+    dataCache = newBuilder()
+        .expireAfterWrite(expireTimeInSeconds, TimeUnit.SECONDS)
+        .build();
+
+    autoExpiringRepeatReadCounterMap = newBuilder()
+        .expireAfterAccess(counterExpireTimeInSeconds, TimeUnit.SECONDS)
+        .build();
+
+    instrumentCacheMetrics();
+  }
+
+  /**
+   * Instrument metrics about this database cache for monitoring, alerting, canary, etc.
+   */
+  private void instrumentCacheMetrics() {
```

Comment:
Can this use MetricReportingCache?

Reply:
It can, you wrote this after me ;P.
- [ ] DRY'ify the code

Reply:
You can't foresee the future?